### PR TITLE
refactor: remove redundant async signature

### LINKED ISSUE #4832: Remove `async` from DNS builder methods
See https://github.com/libp2p/rust-libp2p/pull/4808#discussion_r1387374305_.

### PR DESCRIPTION
**Pull request by dariusc93**
## Description

Resolves #4832

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: libp2p/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@
 - Use `web-time` instead of `instant`.
   See [PR 5347](https://github.com/libp2p/rust-libp2p/pull/5347).
 
+- Remove redundant async signature.
+  See [PR XXXX](https://github.com/libp2p/rust-libp2p/pull/XXXX).
```
Comment (guillaumemichel) [inline libp2p/CHANGELOG.md]:
```suggestion
  See [PR 5468](https://github.com/libp2p/rust-libp2p/pull/5468).
```

File: libp2p/src/builder/phase/quic.rs [hunk 2]
```diff
@@ -162,7 +162,6 @@ impl<T: AuthenticatedMultiplexedTransport> SwarmBuilder<super::provider::AsyncSt
         self.without_quic()
```
Comment (guillaumemichel) [inline libp2p/src/builder/phase/quic.rs:162]: this function (`with_dns` from `quic.rs`) probably doesn't need to remain `async` either

**Comment by dariusc93** [comment]
Though not related to this PR, seems like `write_after_peer_dropped_stream` failed though nothing changed that should have caused it to failed nor does it fail locally. Maybe just a flaky test?

**Comment by guillaumemichel** [comment]
> Though not related to this PR, seems like `write_after_peer_dropped_stream` failed though nothing changed that should have caused it to failed nor does it fail locally. Maybe just a flaky test?

I have rerun the test, and it passed. So the test is probably flaky.

**Review by guillaumemichel — APPROVED** [review]

File: libp2p/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,9 @@
 - Use `web-time` instead of `instant`.
   See [PR 5347](https://github.com/libp2p/rust-libp2p/pull/5347).
 
+- Remove redundant async signature.
```
Comment (jxs) [inline libp2p/CHANGELOG.md]:
```suggestion
- Remove redundant async signature from builder methods.
```

**Review by jxs — APPROVED** [review]
Thanks Darius!

**Pull request merged**